imod/prepare/hfb.py

```diff
@@ -0,0 +1,219 @@
+from itertools import pairwise
+from typing import TYPE_CHECKING, List, Tuple
+
+from imod.typing import PolygonType
+from imod.util.imports import MissingOptionalModule
+
+if TYPE_CHECKING:
+    import shapely
+else:
+    try:
+        import shapely
+    except ImportError:
+        shapely = MissingOptionalModule("shapely")
+
+
+def _line_to_square_zpolygon(
```

Comment:
would a geometry library make sense containing these functions ? (_line_to_square_zpolygon, linestring_to_square_zpolygons , ...) There are also geometry functions in utilities/hfb and other places in this PR